Editor/Scripts/Analytics/MeshsyncAnalytics.cs

```diff
@@ -24,78 +20,23 @@ internal interface IMeshSyncAnalytics {
     }
 
     internal sealed class MeshSyncAnalytics : IMeshSyncAnalytics {
-        private const string VENDORKEY = "unity.meshsync";
-
-        private const string DCCCONFIGURE_INSTALLEDEVENTNAME = "dccConfigure";
-
-        // Data schema version for install event
-        private const int DCCCONFIGURE_VERSION = 4;
-
-        private const string MESHSYNC_SYNC_INSTALLEDEVENTNAME = "meshSync_Sync";
-
-        // Data schema version for sync version
-        private const int MESHSYNC_SYNC_VERSION = 2;
-
-        private const int DCCCONFIGURE_EVENTS_PER_HOUR = 1000;
-        private const int MESHSYNC_SYNC_EVENTS_PER_HOUR = 10000;
-
-        private struct DCCInstallEventData {
-            public string meshSyncDccName;
-        }
-
-        private struct SyncEventData {
-            public string assetSyncType;
-            public string entitySyncType;
-        }
-
-        private static void logIfWarning(AnalyticsResult resp) {
-            if (resp != AnalyticsResult.Ok) {
-                Debug.LogWarning($"Analytics endpoint reported: {resp} when should be {AnalyticsResult.Ok}");
-            }
-        }
-
         static MeshSyncAnalytics() {
-            logIfWarning(
-                EditorAnalytics.RegisterEventWithLimit(
-                    DCCCONFIGURE_INSTALLEDEVENTNAME,
-                    DCCCONFIGURE_EVENTS_PER_HOUR,
-                    10,
-                    VENDORKEY,
-                    DCCCONFIGURE_VERSION));
-
-            logIfWarning(
-                EditorAnalytics.RegisterEventWithLimit(
-                    MESHSYNC_SYNC_INSTALLEDEVENTNAME,
-                    MESHSYNC_SYNC_EVENTS_PER_HOUR,
-                    10,
-                    VENDORKEY,
-                    MESHSYNC_SYNC_VERSION));
+            AnalyticsHandlerFactory.RegisterHandlers();
         }
 
         public void UserInstalledPlugin(string dccTool) {
-            var data = new DCCInstallEventData { meshSyncDccName = dccTool.ToLower() };
-            logIfWarning(EditorAnalytics.SendEventWithLimit(DCCCONFIGURE_INSTALLEDEVENTNAME, data, DCCCONFIGURE_VERSION));
+            new InstallAnalyticsHandler().Send(dccTool);
         }
 
         public void UserSyncedData(MeshSyncAnalyticsData data) {
-            var assetTypeStr = data.assetType == AssetType.Unknown
-                ? "none"
-                : data.assetType.ToString().ToLower();
+            var handler = AnalyticsHandlerFactory.GetHandler(data);
 
-            var entityTypeStr = data.entityType == EntityType.Unknown
-                ? "none"
-                : data.entityType.ToString().ToLower();
-
-            var eventData = new SyncEventData {
-                assetSyncType = assetTypeStr,
-                entitySyncType = entityTypeStr
-            };
+            if (handler == null) {
+                Debug.LogError("No analytics handler found!");
```

Comment:
might be good we add a [MeshSync] prefix here. Also, could we use the MeshSync logger you implemented in another PR to avoid showing this to users?

Reply:
Users should never see this. If this is logged the code needs to be updated to add a handler for the new analytics data.